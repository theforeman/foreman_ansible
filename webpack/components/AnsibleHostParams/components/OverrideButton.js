import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';

import ButtonHelpPopover from './ButtonHelpPopover';

const OverrideButton = ({ toggleField, fieldOverriden, keyId }) => {
  let faIcon;
  let popoverText;
  let popoverId;

  if (!fieldOverriden) {
    faIcon = 'fa-pencil-square-o';
    popoverText = __('Override this value');
    popoverId = `lookup-key-add-override-${keyId}`;
  } else {
    faIcon = 'fa-times';
    popoverText = __('Remove this override');
    popoverId = `lookup-key-remove-override-${keyId}`;
  }

  const button = (
    <Button
      name="button"
      type="button"
      className="btn btn-default btn-md btn-override"
      onClick={toggleField}
    >
      <span className={`fa ${faIcon}`} />
    </Button>
  );

  return (
    <ButtonHelpPopover
      button={button}
      popoverText={popoverText}
      popoverId={popoverId}
    />
  );
};

OverrideButton.propTypes = {
  toggleField: PropTypes.func.isRequired,
  fieldOverriden: PropTypes.bool.isRequired,
  keyId: PropTypes.number.isRequired,
};

export default OverrideButton;
