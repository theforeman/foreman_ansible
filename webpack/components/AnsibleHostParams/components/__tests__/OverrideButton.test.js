import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import OverrideButton from '../OverrideButton';

const keyId = 7;

const toggleField = () => {};

const props = {
  keyId,
  toggleField,
};

const fixtures = {
  'should render when field is overriden': {
    fieldOverriden: true,
    ...props,
  },
  'should render when field is not overriden': {
    fieldOverriden: false,
    ...props,
  },
};

describe('OverrideButton', () =>
  testComponentSnapshotsWithFixtures(OverrideButton, fixtures));
