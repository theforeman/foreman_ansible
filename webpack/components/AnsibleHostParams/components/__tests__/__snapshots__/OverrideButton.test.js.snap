// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverrideButton should render when field is not overriden 1`] = `
<ButtonHelpPopover
  button={
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn btn-default btn-md btn-override"
      disabled={false}
      name="button"
      onClick={[Function]}
      type="button"
    >
      <span
        className="fa fa-pencil-square-o"
      />
    </Button>
  }
  popoverId="lookup-key-add-override-7"
  popoverText="Override this value"
/>
`;

exports[`OverrideButton should render when field is overriden 1`] = `
<ButtonHelpPopover
  button={
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn btn-default btn-md btn-override"
      disabled={false}
      name="button"
      onClick={[Function]}
      type="button"
    >
      <span
        className="fa fa-times"
      />
    </Button>
  }
  popoverId="lookup-key-remove-override-7"
  popoverText="Remove this override"
/>
`;
